hedera-node/src/main/java/com/hedera/services/store/models/Topic.java

```diff
@@ -86,6 +91,46 @@ public static Topic fromGrpcTopicCreate(
 		return topic;
 	}
 
+	/**
+	 * <p>Updates the topic model</p>
+	 * <p><b>Affected properties can be:</b></p>
+	 * <li>Admin key</li>
+	 * <li>Submit key</li>
+	 * <li>Memo</li>
+	 * <li>expirationTimestamp</li>
+	 * <li>autoRenewDurationSeconds</li>
+	 * <li>autoRenewAccountId</li>
+	 * @param op
+	 * 		<p>the Transaction body</p>
+	 * @param designatesAccountRemoval
+	<p>if the auto-renew account is designated for removal, sets the autoRenewAccountId to null.</p>
+	<p>Otherwise - replaces it with the autoRenewAccountId, contained in the transaction body.</p>
+	 */
+	//TODO: Write test for the bellow method and validate, that properly maps the Merkle topic properties to the model
+	public void update(final ConsensusUpdateTopicTransactionBody op, final boolean designatesAccountRemoval) {
```

Comment:
I think that we can mitigate/reduce the penetration of the `GRPC` model to the `TransitionLogic` and pass the exact variables to the method and not the `GRPC` operation. Since they **can** be `null` we should add `@Nullable` annotation